naga/src/valid/expression.rs

```diff
@@ -1332,12 +1334,32 @@ impl super::Validator {
                             _ => return Err(ExpressionError::InvalidArgumentType(fun, 0, arg)),
                         }
                     }
-                    Mf::CountTrailingZeros
-                    | Mf::CountLeadingZeros
+                    // Remove once fixed https://github.com/gfx-rs/wgpu/issues/5276
+                    Mf::CountLeadingZeros
                     | Mf::CountOneBits
                     | Mf::ReverseBits
-                    | Mf::FindLsb
-                    | Mf::FindMsb => {
+                    | Mf::FindMsb
+                    | Mf::FindLsb => {
+                        if arg1_ty.is_some() || arg2_ty.is_some() || arg3_ty.is_some() {
+                            return Err(ExpressionError::WrongArgumentCount(fun));
+                        }
+                        match *arg_ty {
+                            Ti::Scalar(scalar) | Ti::Vector { scalar, .. } => match scalar.kind {
+                                Sk::Sint | Sk::Uint => {
+                                    if scalar.width != 4 {
+                                        return Err(ExpressionError::UnsupportedWidth(
+                                            fun,
+                                            scalar.kind,
+                                            scalar.width,
+                                        ));
+                                    }
+                                }
+                                _ => return Err(ExpressionError::InvalidArgumentType(fun, 0, arg)),
+                            },
+                            _ => return Err(ExpressionError::InvalidArgumentType(fun, 0, arg)),
+                        }
+                    }
+                    Mf::CountTrailingZeros => {
```

Comment:
Considering these work in Metal and HLSL, we should error in the SPIR-V backend instead.

Reply:
I think that's a massive portability hazard - The plan is to polyfill these on spir-v as well (see #5276) so I think it's cleaner to just ban them everywhere until it is available everywhere we support 64bit.

Reply:
That's what we usually do with things that don't yet work in specific backends but since this is an optional native feature I guess validating it for all backends is fine.